lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/driver/ContainerManager.java

```diff
@@ -134,43 +161,227 @@ public void onNext(final Time value) {
       }
     });
 
+    init(rackNames);
+
     LOG.log(Level.FINE, "Initialized Container Manager with {0} containers", capacity);
   }
 
+  private Collection<String> normalize(final Collection<String> rackNames) {
+    return normalize(rackNames, true);
+  }
+
+  /**
+   * Normalizes the rack names.
+   *
+   * @param rackNames
+   *          the rack names to normalize
+   * @param validateEnd
+   *          if true, throws an exception if the name ends with ANY (*)
+   * @return a normalized collection
+   */
+  private Collection<String> normalize(final Collection<String> rackNames,
+      final boolean validateEnd) {
+    final List<String> normalizedRackNames = new ArrayList<>();
```

Comment:
You can pre-allocate the size of the `ArrayList` to be the size of the `Collection` passed.